pennylane/pauli/pauli_arithmetic.py

```diff
@@ -592,6 +631,34 @@ def __truediv__(self, other):
             f"PauliSentence can only be divided by numerical data. Attempting to divide by {other} of type {type(other)}"
         )
 
+    def commutator(self, other):
+        """Compute commutator between two Pauli sentences by iterating over each sentence and commuting
+        the Pauli words pair-wise"""
+        final_ps = PauliSentence()
+
+        if not isinstance(other, PauliSentence):
+            other = qml.pauli.pauli_sentence(other)
+
+        if len(self) == 0:
+            return copy(other)
+
+        if len(other) == 0:
+            return copy(self)
+
+        for pw1 in self:
+            for pw2 in other:
+                comm_pw, coeff = pw1._commutator(pw2)
+                if comm_pw is not None:
```

Comment:
`_commutator` never returns `None`

```suggestion
                if len(comm_pw) != 0:
```

Reply:
Oh good catch! A relic from an old version :)